include/swift/AST/Attr.h

```diff
@@ -2085,6 +2085,26 @@ class TypeSequenceAttr : public DeclAttribute {
   }
 };
 
+/// The @_unavailableFromAsync attribute, used to make function declarations
+/// unavailable from async contexts.
+class UnavailableFromAsyncAttr : public DeclAttribute {
+public:
+  UnavailableFromAsyncAttr(StringRef Message, SourceLoc AtLoc,
+                           SourceRange Range, bool Implicit)
+      : DeclAttribute(DAK_UnavailableFromAsync, AtLoc, Range, Implicit),
+        Message(Message) {}
```

Comment:
Should we copy the message text into the ASTContext so there's no chance of a dangling pointer?

Reply:
I was matching what `AvailableAttr` does for its message. I believe this is safe since the source of the reference is stored in the source manager for the duration of the compilation when parsed from source.
I'm having a little more trouble proving to myself that it's persistent when deserialized from a module file, but if the `@available(message: "foobar", renamed: "blarpy(_:)")` works, then this _should_ be okay. Thoughts? Or does `@available` need fixing too?

Reply:
`AvailableAttr` has been doing this for years, so you're right, it should be fine. Thank you!